src/math/integerhistogram1d.cpp

```diff
@@ -0,0 +1,221 @@
+    std::optional<int> expectedMinimum = bins_.empty() ? std::nullopt : std::optional<int>(bins_.begin()->first);
+    if (minimum_)
+        expectedMinimum = expectedMinimum ? std::min(*minimum_, *expectedMinimum) : *minimum_;
+
+    std::optional<int> expectedMaximum = bins_.empty() ? std::nullopt : std::optional<int>(std::prev(bins_.end())->first);
+    if (maximum_)
+        expectedMaximum = expectedMaximum ? std::max(*maximum_, *expectedMaximum) : *maximum_;
+
+    // Need to check on null values - set to zeroes if both null, or if one is null set it to match the other
+    if (!expectedMinimum && !expectedMaximum)
+    {
+        expectedMinimum = 0;
+        expectedMaximum = 0;
+    }
+    else if (!expectedMinimum)
+        expectedMinimum = expectedMaximum;
+    else if (!expectedMaximum)
+        expectedMaximum = expectedMinimum;
+
+    auto expectedNBins = (*expectedMaximum - *expectedMinimum) + 1;
+
+    // Set up data
+    data.initialise(expectedNBins, true);
+    auto x = *expectedMinimum;
+    for (auto n = 0; n < expectedNBins; n++)
+        data.xAxis(n) = x++;
+    return data;
+}
+
+// Initialise with specified bin range
+void IntegerHistogram1D::initialise(std::optional<int> xMin, std::optional<int> xMax)
+{
+    clear();
+
+    minimum_ = xMin;
+    maximum_ = xMax;
+
+    bins_.clear();
+}
+
+// Zero histogram bins
+void IntegerHistogram1D::zeroBins()
+{
+    bins_.clear();
+    nBinned_ = 0;
+    nMissed_ = 0;
+}
+
+// Return minimum value for data
+std::optional<int> IntegerHistogram1D::minimum() const { return minimum_; }
+
+// Return maximum value for data
+std::optional<int> IntegerHistogram1D::maximum() const { return maximum_; }
+
+// Bin specified value, returning success
+bool IntegerHistogram1D::bin(int x)
+{
+    if ((minimum_ && x < *minimum_) || (maximum_ && x > *maximum_))
+    {
+        nMissed_++;
+        return false;
+    }
+
+    bins_[x].first++;
+
+    return true;
+}
+
+// Return number of values binned over all bins
+long int IntegerHistogram1D::nBinned() const { return nBinned_; }
+
+// Accumulate current histogram bins into averages
+void IntegerHistogram1D::accumulate()
+{
+    for (auto &[key, value] : bins_)
+        value.second += (SampledDouble)value.first;
+
+    // Update accumulated data
+    updateAccumulatedData();
+}
+
+// Return histogram data
+std::map<int, std::pair<long int, SampledDouble>> &IntegerHistogram1D::bins() { return bins_; }
+const std::map<int, std::pair<long int, SampledDouble>> &IntegerHistogram1D::bins() const { return bins_; }
+
+// Add source histogram data into local array
+void IntegerHistogram1D::add(IntegerHistogram1D &other, int factor)
+{
+    if (bins_.size() != other.bins_.size())
+    {
+        Messenger::print("BAD_USAGE - Can't add IntegerHistogram1D data since arrays are not the same size ({} vs {}).\n",
+                         bins_.size(), other.bins_.size());
+        return;
+    }
+    for (auto &[key, value] : bins_)
+
```

Comment:
Just moving the blank line.
```suggestion

    for (auto &[key, value] : bins_)
```